### PR TITLE
Add tests to Module and fix bugs of Lesson

### PR DESCRIPTION
**Pull request by UncleGrandpa925**

### CODE REVIEW

**Comment by UncleGrandpa925** [comment]
Hi @simonteozw I think this PR is good to merge immediately but I just want you to take a look to see what I have done in this PR (mostly for the refactor of Lesson's messages, I wasn't aware that Lesson's messages structure were quite messy)

File: src/main/java/trackitnus/model/ModelManager.java [hunk 1]
```diff
@@ -276,6 +276,11 @@ public void addLesson(Lesson module) {
         trackIter.addLesson(module);
     }
 
+    @Override
+    public boolean canAddModule() {
+        return getFilteredModuleList().size() < NUM_MODULE_LIMIT;
+    }
+
```
Comment (simonteozw) [inline src/main/java/trackitnus/model/ModelManager.java]: I feel that this function should also check if the module alr exists in trackit? Or maybe rename the function to `hasEnoughSpace` to make it clearer

File: src/test/java/trackitnus/logic/commands/module/ModuleCommandTestUtil.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+//    public static void showModuleAtIndex(Model model, Index targetIndex) {
+//        assertTrue(targetIndex.getZeroBased() < model.getFilteredModuleList().size());
+//
+//        Module module = model.getFilteredModuleList().get(targetIndex.getZeroBased());
+//        Predicate<Module> p = t -> t.getName().equals(module.getName());
+//        model.updateFilteredModuleList(p);
+//
+//        assertEquals(1, model.getFilteredModuleList().size());
+//    }
+}
```
Comment (simonteozw) [inline src/test/java/trackitnus/logic/commands/module/ModuleCommandTestUtil.java:32]: Why is this commented out?

**Review by simonteozw — COMMENTED** [review]
Looks good! Just 2 comments

File: src/test/java/trackitnus/logic/commands/module/ModuleCommandTestUtil.java (hunk 2 above)
Comment (UncleGrandpa925) [reply src/test/java/trackitnus/logic/commands/module/ModuleCommandTestUtil.java:32]: Ah this I copied from TaskTestUtil but realised it's not usable yet

**Review by simonteozw — APPROVED** [review]
LGTM

**Pull request merged**